MovingBlocks/Terasology
# Vignette node draft

### Contains

**This PR proposes a solution for issue #3578 by placing vignette at the end of the shader pipeline**, as opposed to the present state where the vignette is a part of initial post-processing node/shader. This solution comes to mind as the blur effects responsible for this issue are currently the very last.
**Whether this approach is correct is a subject for further discussion. It is debatable, whether such change is for the better.**

This PR also attacks Stage 1 of issue #3040 by creating a separate node for Vignette. I would like to ask @vampcat to check if my approach is headed the right way and, if need be, to tag other contributors responsible for such changes to review.

The changes should be rather light-weight, but this is the first time I played with the engine's dag, so keep that in mind :).

### How to test
Enable Vignette, Blur,...various graphical effects combinations to check if the graphics behave correctly. 


## Notes
Suggestion of directly tinting vignette by general composition and tint nodes (currently non-existent) is irrelevant at this stage as vignette node creation is a common step.

### Outstanding before merging

- [ ] Needs to be updated to work with #3732 
- [x] Need to discuss whether the solution to #3578 is wanted. It is debatable whether such change is for the better. @Cervator said whichever looked better to us
- [ ] finish at least stage 1 of #3040  -  enable setting tint from outside somehow
- [ ] verify graphical correctness (correct buffers are passed - hdr, default, no effects are left behind)
**If fix for #3578 is accepted like this:**
- [ ] ~~Move debugging into vignette node? - just to not forget about it until the overhaul is ready~~
- [x]  Move screenshotting from finalPostProcessNode/vignetteNode to OutputNodes

suggested [DAG here](https://sketchboard.me/TBrKdjFknbZv#/)

**VR output node is named OutputToHMDNode, not outputToVRFrameBufferNode**
DAG before:
![image](https://user-images.githubusercontent.com/5579661/52526466-52215980-2cb9-11e9-9018-c96690a48419.png)

DAG after:
![image](https://user-images.githubusercontent.com/5579661/54921504-36d77a00-4f06-11e9-8288-9ed7654db580.png)




please suggest further needs or dismiss this PR as not-important atm.

**Blurred vignette**
![vignette-blurred](https://user-images.githubusercontent.com/5579661/52167150-11549e00-2717-11e9-85e5-d7f1e369bdb7.gif)

**Not blurred vignette**
![vignette-notblurred](https://user-images.githubusercontent.com/5579661/52167145-01d55500-2717-11e9-8553-22d5de42d9b0.gif)
